inference-engine/tests/functional/plugin/conformance/subgraphs_dumper/src/main.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <fstream>
+#include <regex>
+
+#include "inference_engine.hpp"
+
+#include "common_test_utils/file_utils.hpp"
+
+#include "ops_cache.hpp"
+#include "gflag_config.hpp"
+
+// TODO: Poor exceptions handling
+int main(int argc, char *argv[]) {
+    uint8_t ret_code = 0;
+
+
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        return 0;
+    }
+    auto ie = InferenceEngine::Core();
+
+    auto cache = SubgraphsDumper::OPCache::make_cache();
+
+    if (!CommonTestUtils::directoryExists(FLAGS_input_folders)) {
+        std::string msg = "Input directory (" + FLAGS_input_folders + ") is not exist!";
+        throw std::runtime_error(msg);
+    }
+
+    if (!CommonTestUtils::directoryExists(FLAGS_output_folders)) {
+        std::string msg = "Output directory (" + FLAGS_output_folders + ") is not exist!";
+        throw std::runtime_error(msg);
+    }
+
+    std::vector<std::string> input_folder_content;
+    CommonTestUtils::directoryFileListRecursive(FLAGS_input_folders, input_folder_content);
+
+    for (const auto &file : input_folder_content) {
+        try {
+            if (CommonTestUtils::fileExists(file) && std::regex_match(file, std::regex(R"(.*\.xml)"))) {
```

Comment:
1) New standard will be welcome here https://en.cppreference.com/w/cpp/string/basic_string/ends_with :)
2) Creating regex each time might be expensive 
3) e.g.
```
bool ends_with(const std::string& s, const std::string& test){
    return s.size() >= test.size() && std::equal(test.rbegin(), test.rend(), s.rbegin()); //TODO case sensitive maybe?
}

Reply:
Yeah, new standards provide a lot of cool and useful features but for OS compatibility we are still should to use C++11 AFAIK
Regarding regex re-creation I agree, will fix it

Reply:
Fixed